src/messagingsession/MessagingSession.ts

```diff
@@ -9,6 +9,11 @@ export default interface MessagingSession {
    */
```

Comment:
I don't think we need to introduce a new API `startAsync` because the messaging session invokes `messagingSessionDidStart` after receiving the `SESSION_ESTABLISHED` message. i.e., Even if the `startAsync` execution ends, the messaging session has not yet started. What do you think?

Reply:
Good point! @dpwspoon could you please check this? We do not have to await the new endpointUrl resolving which should help with existing `start` API.

Reply:
managed to work around them with some timeouts...

Reply:
messagingSessionDidStart is not entirely true cause there are some assumptions executed by unit tests of calling start and then stop with out listening to the observer... hence the timeouts added to those tests